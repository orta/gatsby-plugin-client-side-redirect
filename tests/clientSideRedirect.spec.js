const clientSideRedirect = require('../clientSideRedirect');

describe('clientSideRedirect', () => {
  it('wraps path in forward slashes', () => {
    expect(clientSideRedirect('toPath')).toMatchSnapshot();
  });

  it('allows existing leading and trailing forward slashes', () => {
    expect(clientSideRedirect('/toPath/')).toMatchSnapshot();
  });

  it('trims leading and trailing whitespace', () => {
    expect(clientSideRedirect(' toPath ')).toMatchSnapshot();
  });

  it('handles deep paths', () => {
    expect(clientSideRedirect('a/b/c/d')).toMatchSnapshot();
  });

  it('handles offset wrapping forward slashes', () => {
    expect(clientSideRedirect('a/b/c/')).toMatchSnapshot();
  });

  it('replaces duplicate slashes with single slash', () => {
    expect(clientSideRedirect('topath//a')).toMatchSnapshot();
  });

  it('leaves full urls untouched', () => {
    expect(clientSideRedirect('http://example.com')).toMatchSnapshot();
    expect(clientSideRedirect('http://example.com/')).toMatchSnapshot();
    expect(clientSideRedirect('http://example.com/a/b/c')).toMatchSnapshot();
  });

  it('handles redirecting to root', () => {
    expect(clientSideRedirect('/')).toMatchSnapshot();
  });

  it('handles redirecting to a file', () => {
    expect(clientSideRedirect('/test.txt')).toMatchSnapshot();
  });

  it('handles redirecting to a file in a folder', () => {
    expect(clientSideRedirect('a/b/test.txt')).toMatchSnapshot();
  });

  it('handles not using xyz/index.html when the file ends with .html', () => {
    expect(clientSideRedirect('a/b/c/d.html')).toEqual(`<script>window.location.href="/a/b/c/d.html"</script>`)
    expect(clientSideRedirect('a/b/c/index.html')).toEqual(`<script>window.location.href="/a/b/c/index.html"</script>`)
    expect(clientSideRedirect('a/b.html')).toEqual(`<script>window.location.href="/a/b.html"</script>`)
  });

});
