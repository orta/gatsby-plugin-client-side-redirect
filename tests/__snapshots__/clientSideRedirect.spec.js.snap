// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clientSideRedirect allows existing leading and trailing forward slashes 1`] = `"<script>window.location.href=\\"/toPath\\"</script>"`;

exports[`clientSideRedirect handles deep paths 1`] = `"<script>window.location.href=\\"/a/b/c/d\\"</script>"`;

exports[`clientSideRedirect handles offset wrapping forward slashes 1`] = `"<script>window.location.href=\\"/a/b/c\\"</script>"`;

exports[`clientSideRedirect handles redirecting to a file 1`] = `"<script>window.location.href=\\"/test.txt\\"</script>"`;

exports[`clientSideRedirect handles redirecting to a file in a folder 1`] = `"<script>window.location.href=\\"/a/b/test.txt\\"</script>"`;

exports[`clientSideRedirect handles redirecting to root 1`] = `"<script>window.location.href=\\"/\\"</script>"`;

exports[`clientSideRedirect leaves full urls untouched 1`] = `"<script>window.location.href=\\"http://example.com\\"</script>"`;

exports[`clientSideRedirect leaves full urls untouched 2`] = `"<script>window.location.href=\\"http://example.com/\\"</script>"`;

exports[`clientSideRedirect leaves full urls untouched 3`] = `"<script>window.location.href=\\"http://example.com/a/b/c\\"</script>"`;

exports[`clientSideRedirect replaces duplicate slashes with single slash 1`] = `"<script>window.location.href=\\"/topath/a\\"</script>"`;

exports[`clientSideRedirect trims leading and trailing whitespace 1`] = `"<script>window.location.href=\\"/toPath\\"</script>"`;

exports[`clientSideRedirect wraps path in forward slashes 1`] = `"<script>window.location.href=\\"/toPath\\"</script>"`;
